kafka/src/main/java/org/jboss/aerogear/unifiedpush/kafka/streams/NotificationRouterStreamsHook.java

```diff
@@ -0,0 +1,198 @@
+
+import org.jboss.aerogear.unifiedpush.api.FlatPushMessageInformation;
+import org.jboss.aerogear.unifiedpush.api.PushApplication;
+import org.jboss.aerogear.unifiedpush.api.Variant;
+import org.jboss.aerogear.unifiedpush.api.VariantType;
+import org.jboss.aerogear.unifiedpush.kafka.KafkaClusterConfig;
+import org.jboss.aerogear.unifiedpush.kafka.serdes.InternalUnifiedPushMessageSerde;
+import org.jboss.aerogear.unifiedpush.kafka.serdes.PushApplicationSerde;
+import org.jboss.aerogear.unifiedpush.message.InternalUnifiedPushMessage;
+import org.jboss.aerogear.unifiedpush.message.holder.MessageHolderWithVariants;
+import org.jboss.aerogear.unifiedpush.service.GenericVariantService;
+import org.jboss.aerogear.unifiedpush.service.metrics.PushMessageMetricsService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+import static org.jboss.aerogear.unifiedpush.api.VariantType.*;
+
+/**
+ * Class that will be initialized on startup, which reads messages from the input topic
+ * {@link KafkaClusterConfig#NOTIFICATION_ROUTER_STREAMS_INPUT_TOPIC} containing key/value pair (PushApplication, InternalUnifiedPushMessage).
+ *
+ * These records are split into subrecords, based on the message's variants. This allows messages to be processed separately,
+ * giving attention to limitations and requirements of specific push networks.
+ *
+ * Once processing is performed, records are streamed to output topics based on their variant type for further processing
+ * in {@link org.jboss.aerogear.unifiedpush.message.token.TokenLoader}.
+ */
+@Singleton
+@Startup
+public class NotificationRouterStreamsHook {
+
+    private final static Logger logger = LoggerFactory.getLogger(NotificationRouterStreamsHook.class);
+
+    private KafkaStreams streams;
+
+    private final String STREAMS_INPUT_TOPIC = KafkaClusterConfig.NOTIFICATION_ROUTER_STREAMS_INPUT_TOPIC;
+
+    public final String ADM_TOPIC = "agpush_admPushMessageTopic";
+
+    public final String ANDROID_TOPIC = "agpush_gcmPushMessageTopic";
+
+    public final String IOS_TOPIC = "agpush_apnsPushMessageTopic";
+
+    public final String SIMPLE_PUSH_TOPIC = "agpush_simplePushMessageTopic";
+
+    public final String WINDOWS_MPNS_TOPIC = "agpush_mpnsPushMessageTopic";
+
+    public final String WINDOWS_WNS_TOPIC = "agpush_wnsPushMessageTopic";
+
+
+    @Resource(name = "DefaultManagedExecutorService")
+    private ManagedExecutorService executor;
+
+    @Inject
+    private Instance<GenericVariantService> genericVariantService;
+
+    @Inject
+    private PushMessageMetricsService metricsService;
+
+    /**
+     * Records of type (PushApplication, InternalUnifiedPushMessage) are split into subrecords based on the message's
+     * variants.
+     *
+     * Each subrecord is transformed into a record with key/value pair (InternalUnifiedPushMessage, Variant)
+     * and streamed to an output topic based on its variant type.
+     */
+    @PostConstruct
+    private void startup() {
```

Comment:
Can we use something like: ` public void init(@Observes @Initialized(ApplicationScoped.class) Object init)` instead of `PostConstruct` ?